common/spring-cloud-stream-modules-common-configuration/src/main/java/org/springframework/cloud/stream/script/ScriptModulePropertiesFactoryBean.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.script;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import org.springframework.beans.factory.config.PropertiesFactoryBean;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * @author David Turanski
+ * @author Gary Russell
+ */
+public class ScriptModulePropertiesFactoryBean extends PropertiesFactoryBean {
```

Comment:
I wonder why existing `ScriptVariableProperties` and `ScriptVariableGeneratorConfiguration` don't fit your requirements.
From other side if that, why don't improve those, rather than introduce a new abstraction.

And now I see that everything around `script` should be refactored.
See `GroovyTransformProcessorProperties` and `GroovyFilterProcessorProperties` as well as their usage of `ScriptVariableGeneratorConfiguration`.